patches/ios-chrome-browser-https_upgrades-model-https_only_mode_upgrade_tab_helper.mm.patch

```diff
@@ -0,0 +1,22 @@
+diff --git a/ios/chrome/browser/https_upgrades/model/https_only_mode_upgrade_tab_helper.mm b/ios/chrome/browser/https_upgrades/model/https_only_mode_upgrade_tab_helper.mm
+index 052275cb0b38783f07f1ba05ce9f801178223f9c..7797f9c93bde5649d15fca2c0984b6816c486b2c 100644
+--- a/ios/chrome/browser/https_upgrades/model/https_only_mode_upgrade_tab_helper.mm
++++ b/ios/chrome/browser/https_upgrades/model/https_only_mode_upgrade_tab_helper.mm
+@@ -270,7 +270,7 @@ void HttpsOnlyModeUpgradeTabHelper::ShouldAllowResponse(
+   // Upgrade to HTTPS if the navigation wasn't upgraded before. Ignore POST
+   // navigations.
+   if (item_pending->GetHttpsUpgradeType() == web::HttpsUpgradeType::kNone &&
+-      !navigation_is_post_) {
++      !navigation_is_post_ BRAVE_EXCLUDE_BACKFORWARD_NAVS) {
```

Comment:
chromium doesn't cover back/forward navigations? Since it was already upgraded on the original navigation, shouldn't it skip this? `Upgrade to HTTPS if the navigation wasn't upgraded before`

Reply:
They dont cover this in their tab helper properly, no.

Reply:
is that an upstream bug?

Reply:
I believe so, yes. I had a chat with them in the Chromium Slack a while back and they confirmed they still had flows which triggered DCHECKs for this specific usage (which is currently disabled by default)

Reply:
can we add a unit test that covers this? There is `HttpsOnlyModeUpgradeTabHelperTest` so should be easy to create our own?

Reply:
We currently dont support running Chromium unit tests on iOS

Reply:
not chromium unit tests, I'm saying to create our own unit test based on that one

Reply:
This would still be a chromium based test I mean, using the chromium test infrastructure to setup and test a tab helper

Reply:
I see, do we have an issue open to get unit tests working?